spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/AbstractServerConnectionFactory.java

```diff
@@ -198,4 +201,29 @@ protected void publishServerExceptionEvent(Exception e) {
 		}
 	}
 
+	protected void publishServerListeningEvent(int port) {
+		if (getApplicationEventPublisher() != null) {
+			TcpConnectionServerListeningEvent event = new TcpConnectionServerListeningEvent(this, port);
+			TaskScheduler taskScheduler = this.getTaskScheduler();
+			if (taskScheduler != null) {
+				try {
+					taskScheduler.schedule(new Runnable() {
+
+						@Override
+						public void run() {
+							getApplicationEventPublisher().publishEvent(event);
+						}
+
+					}, new Date());
+				}
+				catch (TaskRejectedException e) {
+					getApplicationEventPublisher().publishEvent(event);
+				}
+			}
+			else {
+				getApplicationEventPublisher().publishEvent(event);
```

Comment:
Why have you started to use `taskScheduler`?
I understand the `async` wish, but that is the first time in our code...
Anyway I think it would be better to extract variable for the `getApplicationEventPublisher()`. Or does compiler do the stuff for us?